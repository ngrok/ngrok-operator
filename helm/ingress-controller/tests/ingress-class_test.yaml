suite: test ingress-class
templates:
- ingress-class.yaml
tests:
- it: Should match snapshot
  asserts:
  - matchSnapshot: {}
- it: Creates an default ingress class called ngrok by default
  asserts:
  - isKind:
      of: IngressClass
  - hasDocuments:
      count: 1
- it: Does not create an ingress class when ingressClass.create is false
  set:
    ingressClass.create: false
  asserts:
  - hasDocuments:
      count: 0
