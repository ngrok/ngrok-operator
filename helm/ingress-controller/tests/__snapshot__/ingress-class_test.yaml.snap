Should match snapshot:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: IngressClass
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-ingress-controller
        app.kubernetes.io/part-of: kubernetes-ingress-controller
        app.kubernetes.io/version: 0.4.0
        helm.sh/chart: kubernetes-ingress-controller-0.6.0
      name: ngrok
    spec:
      controller: k8s.ngrok.com/ingress-controller
